homeassistant/components/zamg/config_flow.py

```diff
@@ -33,25 +34,34 @@ async def async_step_user(
             self._client.session = async_get_clientsession(self.hass)
 
         if user_input is None:
-            closest_station_id = await self._client.closest_station(
-                self.hass.config.latitude,
-                self.hass.config.longitude,
-            )
-            LOGGER.debug("config_flow: closest station = %s", closest_station_id)
-            stations = await self._client.zamg_stations()
-            user_input = {}
-
-            schema = vol.Schema(
-                {
-                    vol.Required(CONF_STATION_ID, default=closest_station_id): vol.In(
-                        {
-                            station: f"{stations[station][2]} ({station})"
-                            for station in stations
-                        }
-                    )
-                }
-            )
-            return self.async_show_form(step_id="user", data_schema=schema)
+            try:
+                stations = await self._client.zamg_stations()
+                closest_station_id = await self._client.closest_station(
+                    self.hass.config.latitude,
+                    self.hass.config.longitude,
+                )
+                LOGGER.debug("config_flow: closest station = %s", closest_station_id)
+                user_input = {}
+
+                schema = vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_STATION_ID, default=closest_station_id
+                        ): vol.In(
+                            {
+                                station: f"{stations[station][2]} ({station})"
+                                for station in stations
+                            }
+                        )
+                    }
+                )
+                return self.async_show_form(step_id="user", data_schema=schema)
+            except (ZamgApiError, ZamgNoDataError, ValueError) as err:
+                LOGGER.error("Config_flow: Received error from ZAMG: %s", err)
+                errors["base"] = "cannot_connect"
```

Comment:
We don't need to update `errors` if we abort. `errors` are only for showing on the form.

Reply:
But here we are in the user form.
Or did I get it wrong?

Reply:
the next thing that happens here is 
```py
return self.async_abort(
                     reason="cannot_connect", description_placeholders=errors
                 )
```
where the `reason` is already translated https://github.com/home-assistant/core/blob/2c4f9ca0374d5740c4708d53077271cb9a78ac7f/homeassistant/components/zamg/strings.json#L17 and the `placeholder` argument will not be used.

Same in line 77

`errors` dict is used eg. for showing a form again after validation failed - in `self.async_show_form()`

Reply:
Ok,
so you mean if the `reason` is fully translated, we don't need the `description_placeholders` at all?
In our case here we don't have any `reason` with placeholders.

Then we can get rid of the `errors`, right?

Reply:
Exactly. `errors` is not used in that case.

Reply:
Not resolved.

Reply:
Finally it should be resolved.